src/components/login_page_body/login_page_body.beta.tsx

```diff
@@ -43,78 +41,91 @@ const LoginPageBody = () => {
     }
   };
 
-  // 新增：呼叫後端 signInAPI
-  const callLogInAPI = async (authWith: AuthWith, oauthResponse: SignInResponse | undefined) => {
-    try {
-      const response = await logInTrigger({
-        body: { authWith, oauthResponse },
-      });
-
-      // 處理後端返回的 user 和 hasReadAgreement 資料
-      const { success, data } = response;
-      if (success && data) {
-        const { user, hasReadAgreement } = data;
-        if (!hasReadAgreement && !isAgreeWithInfomationConfirmModalVisible) {
-          // 如果用戶尚未同意條款，顯示同意條款的模態框
-          TOSNPrivacyPolicyConfirmModalCallbackHandler(() => handleUserAgree(authWith));
+  useEffect(() => {
+    // eslint-disable-next-line no-console
+    console.log('session:', session);
+    if (status === 'unauthenticated') {
+      try {
+        if (typeof update === 'function') {
+          update();
+        }
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('Session update failed:', error);
+        // 可能需要跳轉到錯誤頁面或顯示錯誤消息
+      }
+    }
+    // eslint-disable-next-line no-console
+    console.log(
+      `status: ${status}, isAgreeWithInfomationConfirmModalVisible: ${isAgreeWithInfomationConfirmModalVisible}, user.hasReadAgreement: ${data?.user?.hasReadAgreement}`
+    );
+    if (status === 'authenticated') {
+      const user = data?.user;
+      if (user && user?.hasReadAgreement) {
+        if (isAgreeWithInfomationConfirmModalVisible) {
           agreeWithInfomationConfirmModalVisibilityHandler();
-        } else {
-          // 用戶已經同意條款，直接進行登入後的邏輯處理。TODO: (20240812-Tzuhan) route to select-company page
-          // eslint-disable-next-line no-console
-          console.log('用戶已登入:', user);
-          router.push(ISUNFA_ROUTE.SELECT_COMPANY);
         }
-      } else {
+        // 用户已同意條款，跳轉到公司選擇頁面
```

Comment:
- The deprecation time needs to be annotated
- Annotation format does not comply